src/server/scripts/Kalimdor/BlackfathomDeeps/boss_kelris.cpp

```diff
@@ -47,10 +48,23 @@ class boss_kelris : public CreatureScript
     {
         boss_kelrisAI(Creature* creature) : BossAI(creature, DATA_KELRIS) { }
 
+        void Reset() override
+        {
+            _Reset();
+            DoCast(me, SPELL_BLACKFATHOM_CHANNELING, true);
+        }
+
+        void JustReachedHome() override
+        {
+            _JustReachedHome();
+            DoCast(me, SPELL_BLACKFATHOM_CHANNELING, true);
```

Comment:
this looks a hacky.
If i remember right it will work like:

Evade -> JustReachedHome -> Reset

When it happens, SPELL_BLACKFATHOM_CHANNELING will called twice.
And still, dont use DoCast(me, xxxx, ?true/false);
use: DoCastSelf(xxx) method.

Reply:
yes it should cast that - but it actually does not, therefore i added it here

will update the DoCastSelf

Reply:
You are treating the symptom not the disease

Reply:
oh i just see

void BossAI::_JustReachedHome()
{
    me->setActive(false);
}

so no reset call?

Reply:
Hmm, someone removed It ?
Put a breakpoint in debug mode to be sure.

Reply:
yep just tested. reset is called once the creature starts evading, but not again once the creature reaches the homeposition

Reply:
So still has the Double call

Reply:
yep - sai in general has more calls and connections to other functions than bossai - they are not synchronous

Reply:
@Keader here https://imgur.com/a/a6qL2Ng the spell is cast on reset, but it get interrupted because the reset is called when the boss starts evading and not when it reach home position.

Reply:
you guys checked on sniffs, if has no delay in cast?
because... to me, looks like just missing a little event (eventmap) to handle with it.
and still is weird... Reset was called when creature reachs home position, not before, when it changed?

Reply:
I don't know, but the script of this boss doesn't really matter (also I checked sniffs and it just channel the spell)
What matter is fixing the reset for (ScriptedAI, BossAI, WorldBossAI)

Reply:
yep i would suggest to merge the functions that are called from sai to worldbossai and bossai. sai works the best for event stuff

Reply:
It must be implemented in ScriptedAI as worldbossai and bossai inherit every thing from there.